service-base/src/main/java/org/eclipse/hono/service/credentials/CompleteBaseCredentialsService.java

```diff
@@ -111,7 +111,7 @@ protected CompleteBaseCredentialsService(final HonoPasswordEncoder pwdEncoder) {
         try {
             payload.checkValidity(this::checkSecret);
             final Future<CredentialsResult<JsonObject>> result = Future.future();
-            add(tenantId, JsonObject.mapFrom(payload), result.completer());
+            add(tenantId, request.getJsonPayload(), result.completer());
```

Comment:
The *processAddRequest* creates the CredentialsObject instance from the payload and then invokes *hashPlainPasswords()* with the CredentialsObject. So, with your change, we would be losing the password hashes that have been created from any plain text passwords contained in the original payload.
A fix will probably require a change to `CredentialsObject` to also allow additional properties to be included ...

Reply:
Sorry, I submitted that too fast, should have thought of that.
we could also insert the hashed password in the Json object, in the end the DR expect JSON, so why not stick to it ? having extra properties is very easy.